target/linux/ipq806x/image/Makefile

```diff
@@ -328,6 +328,18 @@ define Device/tplink_vr2600v
 endef
 TARGET_DEVICES += tplink_vr2600v
 
+define Device/ubnt_unifi-ap-hd
+	$(call Device/FitImageLzma)
+	DEVICE_VENDOR := Ubiquiti
+	DEVICE_MODEL := UniFi AP HD
+	SOC := qcom-ipq8064
+	BLOCKSIZE := 64k
+	IMAGE_SIZE := 14784k
+	DEVICE_PACKAGES := ath10k-firmware-qca9984-ct
+	IMAGE/sysupgrade.bin := append-kernel | append-rootfs | pad-rootfs | append-metadata | check-size
```

Comment:
Since the device has dropbear running with root access in vendor firmware I tried writing the sysupgrade image to the firmware partition using dd. Worked, but OpenWrt does not start with a kernel panic. Is there some step missing for the sysupgrade image chain?
```
Starting kernel ...

[    0.728129] /dev/root: Can't open blockdev
[    0.728306] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    0.731126] CPU1: stopping
...
```

Reply:
most likely mtdsplit cannot detect rootfs because its not starting on a new block...
aka its a padding issue

try this
```
	IMAGE/sysupgrade.bin := append-kernel | pad-to 64k | append-rootfs | pad-rootfs | append-metadata | check-size
```

Reply:
Even with added padding, I still get the same kernel panic. squashfs starts at 2949120 which aligns to 64k blocks.

Reply:
Please use `pad-to $$$$(BLOCK_SIZE)` instead of the literal 64k.

Reply:
what if you do a sysupgrade while booted from TFTP with initramfs-kernel.bin

Reply:
> Please use `pad-to $$$$(BLOCK_SIZE)` instead of the literal 64k.

Done.

> what if you do a sysupgrade while booted from TFTP with initramfs-kernel.bin

Same kernel panic.

Reply:
I'm out of ideas...but can we have a full kernel log for the panic?

also before you do that, can you rebase to current master branch, there was an issue in master for sysupgrades discovered a week ago and fixed days ago I don't know if it applies or not...

`git pull --rebase https://github.com/openwrt/openwrt.git master`

and make sure your next build is clean

`make clean`

Reply:
Rebased on current master yesterday, removed the build directory and checked out a fresh copy of my branch before building. Flashed with `sysupgrade -n` from booted initramfs image, kernel panic is still there. Here is the kernel log:

```
Starting kernel ...

[    0.728623] /dev/root: Can't open blockdev
[    0.728801] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    0.731620] CPU1: stopping
[    0.739861] CPU: 1 PID: 0 Comm: swapper/1 Not tainted 5.4.85 #0
[    0.742547] Hardware name: Generic DT based system
[    0.748393] [<c030f974>] (unwind_backtrace) from [<c030b984>] (show_stack+0x14/0x20)
[    0.753246] [<c030b984>] (show_stack) from [<c08cae60>] (dump_stack+0x94/0xa8)
[    0.761142] [<c08cae60>] (dump_stack) from [<c030eba0>] (handle_IPI+0x184/0x1b8)
[    0.768174] [<c030eba0>] (handle_IPI) from [<c05d4cb0>] (gic_handle_irq+0xb4/0xb8)
[    0.775724] [<c05d4cb0>] (gic_handle_irq) from [<c0301a8c>] (__irq_svc+0x6c/0x90)
[    0.783090] Exception stack(0xdd46bf18 to 0xdd46bf60)
[    0.790647] bf00:                                                       00000000 00000000
[    0.795707] bf20: 1d063000 ddba2a40 dcc5c400 00000000 ddba1e30 00000000 00000000 00000000
[    0.803868] bf40: 2b9b73c0 2b7c1f20 00000015 dd46bf68 c0725ac0 c0725ac4 00000013 ffffffff
[    0.812025] [<c0301a8c>] (__irq_svc) from [<c0725ac4>] (cpuidle_enter_state+0x94/0x498)
[    0.820178] [<c0725ac4>] (cpuidle_enter_state) from [<c0725f0c>] (cpuidle_enter+0x30/0x4c)
[    0.827987] [<c0725f0c>] (cpuidle_enter) from [<c034ac1c>] (do_idle+0x1d8/0x240)
[    0.836311] [<c034ac1c>] (do_idle) from [<c034af2c>] (cpu_startup_entry+0x1c/0x20)
[    0.843863] [<c034af2c>] (cpu_startup_entry) from [<423024cc>] (0x423024cc)
[    0.851234] Rebooting in 1 seconds..
```